csrc/preseg_passes/propagate_shardings.cpp

```diff
@@ -50,7 +51,7 @@ void PropagateShardingsPass::runPass(Fusion* fusion) {
 
     if (tv->hasDeviceMesh()) {
       update_if_null(tv_with_mesh, tv);
-    } else {
+    } else if (!tv->isCpuScalar()) {
```

Comment:
I'd change this to
```
if (tv->isCpuScalar()) {
  continue;
}

if (tv->hasDeviceMesh()) {
  update_if_null(...);
} else {
  update_if_null(...);
}
```
Same logic but slightly easier to follow -- CPU scalars are skipped as a corner case at the beginning.